astropy/io/misc/asdf/tags/coords/tests/test_representation.py

```diff
@@ -0,0 +1,48 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+import pytest
+
+
+import astropy.units as u
+
+from astropy.coordinates import SphericalRepresentation, UnitSphericalRepresentation, Longitude, Latitude
+
+asdf = pytest.importorskip('asdf')
+from asdf.tests.helpers import assert_roundtrip_tree
+
+
+class Longitude180(Longitude):
+
+    def __new__(cls, angle, unit=None, wrap_angle=180 * u.deg, **kwargs):
+        self = super(Longitude180, cls).__new__(cls,
+                                                angle,
+                                                unit=unit,
+                                                wrap_angle=wrap_angle,
+                                                **kwargs)
+        return self
+
+
+class UnitSphericalWrap180Representation(UnitSphericalRepresentation):
+    attr_classes = OrderedDict([('lon', Longitude180), ('lat', Latitude)])
+
+
+def test_unitspherical(tmpdir):
+    tree = {'representation': UnitSphericalRepresentation(10*u.deg, 10*u.deg)}
+    assert_roundtrip_tree(tree, tmpdir)
+
+
+def test_spherical(tmpdir):
+    tree = {'representation': SphericalRepresentation(10*u.deg, 10*u.deg, 2.2*u.AU)}
+    assert_roundtrip_tree(tree, tmpdir)
+
+
+def test_unitspherical180(tmpdir):
+    x = UnitSphericalWrap180Representation(-10*u.deg, 10*u.deg)
+    # Without this here this test fails.
```

Comment:
If anyone more familiar with the insane inner workings of representations (@eteq @adrn ?) has any ideas what might be causing this please let me know!